software/base/src/main/java/brooklyn/entity/basic/SoftwareProcessImpl.java

```diff
@@ -429,8 +432,21 @@ protected final void callStartHooks() {}
             if (PortRange.class.isAssignableFrom(k.getType())) {
                 PortRange p = (PortRange)getConfig(k);
                 if (p != null && !p.isEmpty()) ports.add(p.iterator().next());
+            }   
+        }
+        
+        Map<String, Object> provisioningProperties = getConfig(PROVISIONING_PROPERTIES);
+        if(provisioningProperties.containsKey("required.ports")){
+            Object requiredPorts = provisioningProperties.get("required.ports");
+            if (requiredPorts instanceof Integer){
+                ports.add((Integer)requiredPorts);
+            }else if (requiredPorts instanceof List) {
```

Comment:
Won't a cooma-separated `String` or single `Integer` get coerced to a `List<Integer>` using a `ConfigKey` or at least you can call the appropriate `TypeCoercions#coerce(...)` method?